src/bot.ts

```diff
@@ -174,20 +174,10 @@ export default abstract class BotClient {
    * @returns The BotChannel with the specified ID.
    */
   public getChannelByID(id: string): BotChannel {
-    const channels = getSubscribers()[this.name];
-    const channel = new BotChannel(id, this);
+    const channels = this.getBotChannels();
+    const filteredChannels = channels.filter((channel) => channel.id === id);

```

Comment:
you can also use `Array.find()` for this, which will return -1 if the channel does not already exist